app/controllers/api/base_controller.rb

```diff
@@ -9,6 +9,57 @@ class BaseController < ActionController::API
     rescue_from ActionController::BadRequest, with: :bad_request_response
     rescue_from ArgumentError, with: :bad_request_response
 
+    def append_info_to_payload(payload)
+      super
+      payload[:current_user_class] = current_lead_provider&.class&.name
+      payload[:current_user_id] = current_lead_provider&.id
+      payload[:current_user_name] = current_lead_provider&.name
+
+      payload[:request_headers] = request.env.slice(
+        "CONTENT_TYPE",
+        "GATEWAY_INTERFACE",
+        "HTTP_ACCEPT",
+        "HTTP_ACCEPT_ENCODING",
+        "HTTP_AUTHORIZATION",
+        "HTTP_CONNECTION",
+        "HTTP_HOST",
+        "HTTP_USER_AGENT",
+        "HTTP_VERSION",
+        "ORIGINAL_FULLPATH",
+        "ORIGINAL_SCRIPT_NAME",
+        "PATH_INFO",
+        "QUERY_STRING",
+        "REMOTE_ADDR",
+        "REQUEST_METHOD",
+        "REQUEST_PATH",
+        "REQUEST_URI",
+        "ROUTES_9080_SCRIPT_NAME",
+        "SCRIPT_NAME",
+        "SERVER_NAME",
+        "SERVER_PORT",
+        "SERVER_PROTOCOL",
+        "SERVER_SOFTWARE",
+        "puma.request_body_wait",
+        "rack.after_reply",
+        "rack.attack.called",
+        "rack.attack.throttle_data",
+        "rack.hijack?",
+        "rack.multiprocess",
+        "rack.multithread",
+        "rack.request.query_hash",
+        "rack.request.query_string",
+        "rack.run_once",
+        "rack.url_scheme",
+        "rack.version",
+      ).transform_values(&:to_s).to_json
+      payload[:request_body] = request.raw_post
+
+      payload[:response_headers] = response.headers.transform_values(&:to_s).to_json
+      if response.status != 200
+        payload[:response_body] = response.body
+      end
+    end
```

Comment:
Seems other projects adding `request_query_params` to the payload as well.. maybe it's a good idea:

https://github.com/DFE-Digital/apply-for-teacher-training/blob/99dc097ac0c47678865ff2aa1f14dd3e2701ad81/app/controllers/api_docs/api_docs_controller.rb#L8

Reply:
Think the query_params are mixed into the `params_json` if a request_body is sent.. just wondering if make sense to have a specific field in the payload only for the `query_params`.

![image](https://github.com/user-attachments/assets/e490fd89-5c1d-4a52-aa03-2e6b58922fa7)

Reply:
Ok i've added `query_params`